src/Strategy/Sqlite.hs

```diff
@@ -0,0 +1,179 @@
+{-# LANGUAGE TupleSections #-}
+
+-- Description: Discovery and analysis functions for Sqlite3 backed RPM databases
+module Strategy.Sqlite (
+  -- for testing
+  SqliteDB (..),
+  readSqliteDBPackages,
+  discover,
+) where
+
+import App.Fossa.Analyze.Types (AnalyzeProject (analyzeProject, analyzeProject'))
+import Container.OsRelease (OsInfo (..))
+import Control.Algebra (Has)
+import Control.Effect.Diagnostics (Diagnostics, context, warn)
+import Control.Effect.Lift (Lift, sendIO)
+import Control.Effect.Reader (Reader)
+import Data.Aeson (ToJSON)
+import Data.Bifunctor (first)
+import Data.ByteString qualified as BS
+import Data.ByteString.Lazy qualified as BLS
+import Data.Either (partitionEithers)
+import Data.Foldable (traverse_)
+import Data.Int (Int64)
+import Data.Rpm.DbHeaderBlob (PkgConversionError, PkgInfo (..), pkgInfoToDependency, readPackageInfo)
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import Database.SQLite3 (ColumnIndex (ColumnIndex), StepResult (..), close, columnBlob, columnInt64, finalize, prepare, step)
+import Database.SQLite3 qualified as SQLite
+import Discovery.Filters (AllFilters)
+import Discovery.Simple (simpleDiscover)
+import Discovery.Walk (WalkStep (WalkContinue), findFirstMatchingFile, walkWithFilters')
+import Effect.ReadFS (ReadFS, readContentsBS)
+import GHC.Generics (Generic)
+import Graphing (directs)
+import Path (Abs, Dir, File, Path, parent, toFilePath)
+import Path.IO (withSystemTempFile)
+import Types (DependencyResults (..), DiscoveredProject (..), DiscoveredProjectType (SqliteDBProjectType), GraphBreadth (Complete))
+
+data SqliteDB = SqliteDB
+  { dbDir :: Path Abs Dir
+  , dbFile :: Path Abs File
+  , osInfo :: OsInfo
+  }
+  deriving (Eq, Ord, Show, Generic)
+
+instance ToJSON SqliteDB
+
+discover ::
+  ( Has ReadFS sig m
+  , Has Diagnostics sig m
+  , Has (Reader AllFilters) sig m
+  ) =>
+  OsInfo ->
+  Path Abs Dir ->
+  m [DiscoveredProject SqliteDB]
+discover osInfo = simpleDiscover (findProjects osInfo) mkProject SqliteDBProjectType
+
+findProjects ::
+  ( Has ReadFS sig m
+  , Has (Reader AllFilters) sig m
+  , Has Diagnostics sig m
+  ) =>
+  OsInfo ->
+  Path Abs Dir ->
+  m [SqliteDB]
+findProjects osInfo = walkWithFilters' $ \dir _ files -> do
+  case findFirstMatchingFile
+    [ "Packages.sqlite"
+    , "rpmdb.sqlite"
+    ]
+    files of
```

Comment:
[optional] Keep in mind splitting lines increases cognitive load while reading. Prefer leaving stuff like this on one line or perhaps splitting the list into a separate `where` function to make it easier to read instead of splitting.
```suggestion
  case findFirstMatchingFile ["Packages.sqlite", "rpmdb.sqlite"] files of
```